packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -26,12 +30,13 @@ const getSigningAuthorityOptions = (
       }
     };
   } else if (isPrivateKeys(mnemonicPhraseOrPrivateKeys)) {
-    return {
-      privateKeys: mnemonicPhraseOrPrivateKeys
-    };
+    return typeof mnemonicPhraseOrPrivateKeys === "string" ?
+      { privateKeys: [ mnemonicPhraseOrPrivateKeys ] } :
+      { privateKeys: mnemonicPhraseOrPrivateKeys };
```

Comment:
Instead of this, it's more clear to have a third conditional for `isPrivateKey`.

Going with my suggestion, to define a `LegacyConstructor.Credentials` type:
```typescript
export type Credentials = MnemonicPhrase | PrivateKey | PrivateKey[];
```

Then we'd define three separate type predicates (see [docs about this](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates)):
- `isMnemonic`
- `isPrivateKey`
- `isPrivateKeys`

The idea is that each of these takes a value of any of the three allowed types and returns a boolean based on which one it is. This gets used both at runtime, for conditional logic, and at compile-time, to tell TS not to barf on `for (const privateKey of mnemonicPhraseOrPrivateKeys)` if you wrap it in `if (isPrivateKeys(mnemonicPhraseOrPrivateKeys)) { ... }`.

So then, in this function, we could just lean on having type predicates for each possibility, and move the ternary operator into a new `else if` that's at the same top-level.